File: onlinejudge/leetcode/1431.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = 0;
        for (int candy : candies) {
            max = Math.max(max, candy);
        }

        List<Boolean> answer = new ArrayList<>();
        for (int i = 0; i < candies.length; i += 1) {
            if (candies[i] + extraCandies >= max) {
                answer.add(true);
            } else {
                answer.add(false);
            }
        }
        return answer;
    }
}

File: onlinejudge/leetcode/1431.rb
# @param {Integer[]} candies
# @param {Integer} extra_candies
# @return {Boolean[]}
def kids_with_candies(candies, extra_candies)
  max_candy = candies.max
  answer = []
  candies.each { |candy|
      if candy + extra_candies >= max_candy
          answer.append true
      else
          answer.append false
      end
  }
  return answer
end
